### ISSUE TITLE
Still Unresolved - file_get_contents(...): failed to open stream: No such file or directory

### ISSUE THREAD

**Issue by dbd5** [lajax/yii2-asset-minifier#2]
Hi @lajax ,

I have updated code as necessary but this still throws exactly the same error!
composer didn't seem to pick up 515449d so I had to remove it from composer.json and update - and then add it back and update again.

It seems that the PATH passed to PhpJsMinifier.php by Minifier.php is wrong

The error thrown is;

file_get_contents(/var/www/advanced/frontend/web/vendor/revslider/js/jquery.themepunch.tools.min.js): failed to open stream: No such file or directory

The file you are looking for is actually in;
/var/www/advanced/frontend/web/**assets**/vendor/revslider/js/jquery.themepunch.tools.min.js

Here's the relevant Asset bundle

<?php
namespace frontend\assets;

use yii\web\AssetBundle;
class RevoAsset extends AssetBundle
{
public $basePath = '@webroot';
public $baseUrl = '@web/assets';
public $jsOptions = [
'position' => \yii\web\View::POS_END,
];

public $css = [
    ["css/extralayers.css", "media"=>"screen",],
    ["vendor/revslider/css/settings.css", "media"=>"screen",],
];

public $js = [
    "vendor/revslider/js/jquery.themepunch.tools.min.js",
    "vendor/revslider/js/jquery.themepunch.revolution.min.js",
];

public $images = [
];

public $depends = [
    'frontend\assets\FrontendAsset',
];
}

Is there anything I can do to assist you troubleshoot this bug?

**Comment by moltam** [comment]
Hi,

I think this is not a problem with the extension.

If you specifiy `basePath` as `@webroot`, and a javascript path as `vendor/revslider/js/jquery.themepunch.tools.min.js`, then the full path of the JS file will be `@webroot/vendor/revslider/js/jquery.themepunch.tools.min.js` (`basePath` + JS path), which does not exist, and the extension can't access the file.

Also, why are your files in the `assets` directory? You should not put your files here, as Yii uses this folder for publishing assets. This directory should be ignored in your VCS.

Move your files elsewhere if they are inside `@webroot/assets` (for example `@webroot/css`, `@webroot/js`). Try fix your paths in `$js`, and `$css`. 

Also you could try separate the revslider config into a separate AssetBundle, and specify the bundle in the `depends` attribute (or register in the view, if it used only a few places, not everywhere).

**Comment by dbd5** [comment]
Thanks for clarifying @moltam, I will make these changes and report back.

My intention was not to blame the extension but just to get the help I need. I really do appreciate the effort that has gone into this by @lajax and the many other extensions by the community as well as the volunteer effort to support them.

Adam

**Comment by dbd5** [comment]
Hi @moltam , @lajax - here's my update after applying all recommended changes
- Vendor  folder moved off assets
- Asset Bundle changed to the listing at the end of the post;
- Links to asset files verified working by viewing source.

The error message is now (still in the same function);

**PHP Warning – yii\base\ErrorException

filemtime(): stat failed for /var/www/advanced/frontend/web/js/jquery.themepunch.tools.min.js**
- Please note that it is still looking for the file in the wrong place rather than '@webroot/vendor/revslider' + '/js/jquery.themepunch.revolution.min.js' ($basePath + $js)

---

<?php
namespace frontend\assets;

use yii\web\AssetBundle;
class RevoAsset extends AssetBundle
{

```
public $basePath = '@webroot/vendor/revslider';
public $baseUrl = '@web/vendor/revslider';
public $jsOptions = [
    'position' => \yii\web\View::POS_END,
];

public $css = [
    ["css/extralayers.css", "media"=>"screen",],
    ["css/settings.css", "media"=>"screen",],
];

public $js = [
    "js/jquery.themepunch.tools.min.js",
    "js/jquery.themepunch.revolution.min.js",
];

public $images = [
];

public $depends = [
    'frontend\assets\FrontendAsset',
];
```

}

**Comment by moltam** [comment]
I can't reproduce this with the latest version. However this error happens with 1.0.0 (#4cce94c). Are you sure you are using the latest version? It does not have a version tagged yet, so you have to use `dev-master` in `composer.json`.

**Comment by dbd5** [comment]
Hi @moltam ,

I have now updated with "lajax/yii2-asset-minifier": "dev-master", in composer.json and tried to run again. 
Happy to report that those particular bugs have been resolved in dev-master 8bfd1c7. The extension generated a couple of compressed files in the correct folder but failed again with 2 new errors;
-1- Non support for using arrays to specify js and css options in asset bundles (supported by yii2 since 2.0.7
-2- RuntimeException; Unclosed string at position: 2812; ../JShrink/Minifier.php

I will go ahead and close this and log 2 new issues for proper tracking
